quarkus.jdt.ext/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/scheduler/SchedulerUtils.java

```diff
@@ -0,0 +1,126 @@
+/*******************************************************************************
+* Copyright (c) 2021 Red Hat Inc. and others.
+*
+* This program and the accompanying materials are made available under the
+* terms of the Eclipse Public License v. 2.0 which is available at
+* http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+* which is available at https://www.apache.org/licenses/LICENSE-2.0.
+*
+* SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.microprofile.jdt.internal.quarkus.scheduler;
+
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Pattern;
+
+/**
+ * Utility class for quarkus scheduler
+ *
+ */
+public class SchedulerUtils {
+
+	// Patterns for each cron part of the cron member expression in order of
+	// SchedulerErrorCode enum ordinal
+	// Note: DAY_OF_MONTH and DAY_OF_WEEK are mutually exclusive
+
+	private static final String DAY_OF_MONTH_ABBREV = "^[mM][oO][nN]$|^[tT][uU][eE]$|^[wW][eE][dD]$|^[tT][hH][uU]$|^[fF][rR][iI]$|^[sS][aA][tT]$|^[sS][uU][nN]$";
```

Comment:
It turns out that a cron expression supports 3 character abbreviations of days/months. Not only that, but it supports case mixing too, for example, you can have `jan, JAN, jAn, JAn, etc..` If there's a better way to go about this, please let me know since this regex is quite long.

Reply:
Try https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/regex/Pattern.html#CASE_INSENSITIVE when compiling the expression.